src/mongocrypt.c

```diff
@@ -176,10 +180,13 @@ mongocrypt_setopt_kms_provider_aws (mongocrypt_t *crypt,
                                     int32_t aws_secret_access_key_len)
 {
    mongocrypt_status_t *status;
+   if (!crypt) {
+      return false;
```

Comment:
That;s actually not possible here.  The `CLIENT_ERR` macro expansion only sets the error status if the `status` variable is present and non-NULL.  However, in this function `status` is set from `crypt->status`, which we cannot access until we have checked that `crypt != NULL`.  This is a chicken-and-egg problem and it doesn't seem like there is a clean way to conform to the documented behavior here.  I encountered this in a few other places as well.

Reply:
My mistake; this comment and others concerning a `crypt` parameter should have suggested `BSON_ASSERT_PARAM` of `crypt`. Mention of documentation was to acknowledge that `crypt`-is-null cannot satisfy the observed result described on error; therefore, should be treated as an assertion rather than recoverable error.

Reply:
No worries.  That makes a lot more sense.  I'll convert these instances to use `BSON_ASSERT_PARAM` instead.